velox/type/DecimalUtils.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DecimalUtils.h"
+
+namespace facebook::velox {
+
+std::string formatAsDecimal(const uint8_t scale, int128_t number) {
+  std::stringstream ss;
+  std::string result;
+  if (number == 0)
+    return "0";
+  while (number) {
+    uint64_t remainder = number % 10;
+    number = number / 10;
+    ss << remainder;
+    result.insert(0, ss.str());
+    ss.str("");
+  }
+  std::string formattedStr;
```

Comment:
can we use `fmt::format("{}{}{}",...)` and compute the whole number part, dot, fractional part? If some of these don't exist, they can be empty.

Reply:
Even if we use POWERS_OF_TEN while figuring out the number of leading zeroes after decimal point would require similar steps. For example "0.000123".  unscaled / POWERS[scale] = 0 (which is right). Unscaled % POWERS[scale] =123. This is not exactly what we need. We need to compare the lengths of unscaled and the scale factor to figure out leading zeroes.

Reply:
so 4 parameters?